gem/interpreter.py

```diff
@@ -176,8 +176,18 @@ def _evaluate_mathfunction(e, self):
     names = {"abs": abs,
              "log": math.log}
     op = names[e.name]
-    for idx in numpy.ndindex(result.tshape):
-        result[idx] = op(*(o[o.filter(idx, result.fids)] for o in ops))
+    if e.name is 'imag':
```

Comment:
How could this work? If `e.name` is `'imag'` then line 178 (`op = names[e.name]`) should throw a `KeyError` and execution should not have reached this point...